### PR TITLE
Bulgarian localization

### PR DESCRIPTION
**Pull request by Stoyan83**
Maybe my yml don't have the best structure and I miss to translate some custom messages and alerts but it was a lot of work 😊

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -17,4 +17,19 @@ def configure_permitted_parameters
...
+  end
+  
+  def set_locale
+    I18n.locale = take_locale || I18n.default_locale
+  end
+  
+  def take_locale
+    parsed_locale = params[:locale]
+    I18n.available_locales.map(&:to_s).include?(parsed_locale) ? parsed_locale : nil
+  end
```
Comment (Snick555) [inline app/controllers/application_controller.rb:32]:
```suggestion
  def set_locale
    I18n.locale = I18n.available_locales.include?(params[:locale].to_sym) && params[:locale] || I18n.default_locale
  end
```

File: config/locales/en.yml [hunk 2]
```diff
@@ -30,4 +30,139 @@
 # available at https://guides.rubyonrails.org/i18n.html.
 
 en:
-  hello: "Hello world"
+  confirmation:
```
Comment (Snick555) [inline config/locales/en.yml:33]: for each section could be a different file (`config/locales/en/confirmation.yml`), it's easier to support

File: config/locales/en.yml [hunk 3]
```diff
@@ -30,4 +30,139 @@
...
+    year: "Year"
+    license: "License plate"
+    pic: "Delete picture"
+    delete: "Delete car"
+    back: "Back"
+    charg: "Add Charging"
+    repair: "Add repair"
+    others: "Taxes and others"
+    edit: "Edit your car"
+    e: "Edit "
```
Comment (Snick555) [inline config/locales/en.yml:113]: don't use one-letter naming

File: app/models/charging.rb [hunk 4]
```diff
@@ -1,4 +1,4 @@
 class Charging < ApplicationRecord
   belongs_to :car
-  monetize :amount_cents, numericality: { greater_than: 0 }
+  monetize :amount_cents, numericality: { in: 0.01..4294967295 }
```
Comment (apoltieva) [inline app/models/charging.rb:3]: Since it's in cents it should be in 1.. 4294967295, because 0.01 cent is actually  0.0001 USD...

**Review by apoltieva — APPROVED** [review]

File: config/locales/en.yml (hunk 3 above)
Comment (Stoyan83) [reply config/locales/en.yml]: All done

File: app/models/charging.rb (hunk 4 above)
Comment (Stoyan83) [reply app/models/charging.rb:3]: It's quite right, but in practice it's acting a bit different. When range starts form 1 in the form you are allowed to enter only integers and the model don't accept 0.5 for example.

**Review by apoltieva — APPROVED** [review]

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (Stoyan83) [reply app/controllers/application_controller.rb]: Actually that gives me that error :  
`undefined method `to_sym' for nil:NilClass`
 if the url is 0.0.0.0:3000 it's is working if we specify language 0.0.0.0:3000\en
That way is working better:
` def set_locale
  I18n.locale = I18n.available_locales.map(&:to_sym).include?(params[:locale]) && params[:locale] || I18n.default_locale
  end
  `

**Pull request merged**